core/services/workflows/metering/metering.go

```diff
@@ -128,43 +137,32 @@ func (r *Report) Reserve(ctx context.Context) error {
 
 	rateCard, err := toRateCard(resp.GetRates())
 	if err != nil {
-		r.lggr.Warnf("failed to parse rate card: %s", err)
-		r.enterMeteringMode()
+		r.lggr.Warnf("failed to parse rate card; switching to metering mode: %s", err)
+		r.balance.meteringMode = true
+		r.ready = true
+
 		return nil
 	}
 
 	r.ready = true
-	r.balance = NewBalanceStore(decimal.NewFromFloat32(resp.Credits).IntPart(), rateCard, r.lggr) // TODO remove .IntPart() once balance store uses decimal
-	return nil
-}
-
-func (r *Report) enterMeteringMode() {
-	// TODO: https://smartcontract-it.atlassian.net/browse/CRE-453 pass through errors and persist cause of metering mode on to meteringReport
-	balanceStore := NewBalanceStore(0, map[string]decimal.Decimal{}, r.lggr)
-	balanceStore.AllowNegative()
-	r.ready = true
-	r.balance = balanceStore
-}
+	r.balance = NewBalanceStore(decimal.NewFromFloat32(resp.Credits), rateCard, r.lggr)
 
-// ConvertFromBalance converts a credit amount to a resource dimensions amount.
-func (r *Report) ConvertFromBalance(toUnit string, amount int64) (resources int64, err error) {
-	if !r.ready {
-		return 0, ErrNoReserve
-	}
-	return r.balance.ConvertFromBalance(toUnit, amount), nil
+	return nil
 }
 
 // ConvertToBalance converts a resource dimensions amount to a credit amount.
-func (r *Report) ConvertToBalance(fromUnit string, amount int64) (credits int64, err error) {
+func (r *Report) ConvertToBalance(fromUnit string, amount decimal.Decimal) (decimal.Decimal, error) {
 	if !r.ready {
-		return 0, ErrNoReserve
+		return decimal.Zero, ErrNoReserve
 	}
+
 	return r.balance.ConvertToBalance(fromUnit, amount), nil
 }
 
-// Deduct earmarks an amount of local universal credit balance.
-// We expect to only set this value once - an error is returned if a step would be overwritten.
-func (r *Report) Deduct(ref string, amount int64) error {
+// Deduct earmarks an amount of local universal credit balance. We expect to only set this value once - an error is
+// returned if a step would be overwritten. Deduct expects a NullDecimal because GetMaxSpendForInvocation can return
+// a NullDecimal.
+func (r *Report) Deduct(ref string, amount decimal.NullDecimal) error {
```

Comment:
`NullDecimal` gets handled in the Engine (if not valid, deduct is never called). Could this just be `decimal.Decimal`?

Reply:
That would be simpler, yes